### DISCUSSION TITLE
Slots

### DISCUSSION THREAD

**Discussion by mikev032** [alpinejs/alpine#3191] [1. Help] [answered]
Are slots available in Alpine JS ? Like : https://vuejs.org/guide/components/slots.html?

I am looking to make an repeater, so an alpine component what can repeat the dom inside of itself.

**Comment by ekwoka** [comment]
I'm not sure what you mean by repeater, but fundamentally, the need for "slots" or "children" isn't valid in Alpine, as the component is only the state/methods, and the dom itself is the dom.

Vue has slots because the component dom is abstracted away to another places so you need these options to make the components composable.

So that concept is fundamentally unneeded in Alpine.

So a better understanding of what you mean by repeater is probably needed.

**Comment by SimoTod** [comment, marked as answer]
You probably want to use standard template tags: https://www.w3schools.com/TagS/tag_template.asp

If your logic is well defined, you can abstract it in one or more directives
